src/data/finsupp.lean

```diff
@@ -402,6 +402,19 @@ if_pos rfl
 @[simp] lemma erase_ne [has_zero β] {a a' : α} {f : α →₀ β} (h : a' ≠ a) : (f.erase a) a' = f a' :=
 if_neg h
 
+lemma erase_single [has_zero β] {a : α} {b : β} : (erase a (single a b)) = 0 := begin
+  ext s, classical, by_cases hs : s = a,
+  { rw [hs, erase_same], refl },
+  { rw [erase_ne hs], exact single_eq_of_ne (ne.symm hs) }
+end
+
+lemma erase_single_ne [has_zero β] {a a' : α} {b : β} {hv : a ≠ a'} : (erase a (single a' b)) = single a' b :=
```

Comment:
```suggestion
lemma erase_single_ne [has_zero β] {a a' : α} {b : β} (h : a ≠ a') : (erase a (single a' b)) = single a' b :=
```
The argument `h` should be explicit, because there is no way that Lean can *infer* a proof from the stuff that follows it.